GoogleForCreators/web-stories-wp
# Dashboard: Reposition display on template change in detail view

## Summary
When a new template is selected from the template detail view, the display will reposition to the top of the layout.
![scroll to top of detail view](https://user-images.githubusercontent.com/10720454/93267200-22bdb300-f760-11ea-829f-31e693d70c85.gif)

## Relevant Technical Choices
I wanted to utilize the layout context's action `scrollToTop` to keep this functionality consistent with other spots where we're repositioning the screen within the confines of the Layout context so I refactored this view to be separated into smaller components mirroring explore templates and my stories so that it would be easier to watch for `template` to update with a useEffect hook while having access to the layout context which needed to be _inside_ the layout provider which wasn't going to be super pretty to look at without refactoring. 
While i was in there I wrote a few unit tests and got template details storybook up and running but didn't want to go overboard

_The only things I changed here are the[ code I commented on](https://github.com/google/web-stories-wp/pull/4502/files/dc3e5e219fc8d5097cbeb91c2f948097a34e788f#diff-62ed6123582b767950aeb78dbe21b745), directory and file structure, added storybook, added some unit tests. Did not update the components themselves. Left all functionality as is._

## To-do

N/A

## User-facing changes

Now when you click on a "related" template in the detail grid view the display will reposition to the top of the screen, it will also reposition if you scroll down a bit and use one of the arrows to get a new template in view that way instead of through the related grid. 

## Testing Instructions

Click on a related template in the detail grid view or with the pagination arrows (while scrolled down slightly on browser screen), see that you are brought back to the top of the display.

---



addresses #4306
